### PR TITLE
Refactor PhysicsTest.cpp: add PhysicsManagerTest feature

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

In order to test physics functionality, the scene must first be loaded and initialized. This process would be duplicated for each test. This setup is now abstracted into PhysicsManagerTest feature which can be re-used by each physics test to avoid code duplication.

## How Has This Been Tested

C++ tests still run and pass.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/tests/PhysicsTest.cpp [hunk 1]
```diff
@@ -26,44 +26,63 @@ const std::string physicsConfigFile =
...
-  ResourceManager resourceManager;
+  esp::gfx::WindowlessContext::uptr context_;
+  esp::gfx::Renderer::ptr renderer_;
+
+  ResourceManager resourceManager_;
   SceneManager sceneManager_;
-  std::shared_ptr<PhysicsManager> physicsManager_;
+  PhysicsManager::ptr physicsManager_;
+
+  std::string sceneFile_;
```
Comment (msbaines) [inline src/tests/PhysicsTest.cpp]: Why is this needed? Its not used anywhere.

Comment (aclegg3) [reply src/tests/PhysicsTest.cpp]: 👍 Was used, but no longer necessary. I'll add back later if needed.
